### ISSUE TITLE
[Code]: Is build/StarcoinFramework/BuildInfo.yaml necessary?

### ISSUE THREAD

**Issue by 0xpause** [starcoinorg/starcoin-framework#169]
Is the file `build/StarcoinFramework/BuildInfo.yaml` necessary? It will lower the develop efficiency because that once a PR is merged, all the other PRs have to rebase the main branch manually and run `./scripts/build.sh` to update the file even if no conflicts in `sources` files.

If we remove `build/StarcoinFramework/BuildInfo.yaml`, then the PRs having no conflicts can be merged automaticlly.

**Comment by jolestar** [comment]
So we just keep `build/StarcoinFramework/docs`? @WGB5445 @baichuan3 @jiangying000 @yubing744 @sanlee42

**Comment by yubing744** [comment]
I think Docs can also be deleted, we can generate Docs in github-ci and publish it to github-page, and provide a script in scripts to quickly generate Docs locally.

**Comment by jiangying000** [comment]
> I think Docs can also be deleted, we can generate Docs in github-ci and publish it to github-page, and provide a script in scripts to quickly generate Docs locally.

I'd support this  approach if it does not weaken corrrectness and security, since PR diffs will be more concise

**Comment by WGB5445** [comment]
> I think Docs can also be deleted, we can generate Docs in github-ci and publish it to github-page, and provide a script in scripts to quickly generate Docs locally.

I also agree with this approach. There are too many submission files now. A small change needs to modify the buildfile and exp file at the same time. The build file should not exist in the warehouse.
doc should be placed under wiki
This will be more convenient if we do more optimizations to the documentation system in the future

**Comment by jolestar** [comment]
Ok, let's remove the build dir from the git repo. The primary purpose of keeping the build directory was to detect bytecode changes that the Move or OS versions version changes might bring about.
